### PR TITLE
docker-engine: remove devicemapper dependencies

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/docker-ce-packaging/pull/959

### CODE REVIEW

File: pkg/docker-engine/Dockerfile [hunk 1]
```diff
@@ -188,7 +188,7 @@ ARG DOCKER_ENGINE_REF
 ARG NIGHTLY_BUILD
 WORKDIR /build
 ARG TARGETPLATFORM
-RUN xx-apt-get install -y gcc libc6-dev libapparmor-dev libdevmapper-dev libltdl-dev libseccomp-dev libsecret-1-dev libsystemd-dev libudev-dev pkg-config
+RUN xx-apt-get install -y gcc libc6-dev libapparmor-dev libltdl-dev libseccomp-dev libsecret-1-dev libsystemd-dev libudev-dev pkg-config
```
Comment (thaJeztah) [inline pkg/docker-engine/Dockerfile:191]: interesting actually that we had to manually install these; didn't we let the rpm/deb tools handle install dependencies?

**Review by thaJeztah — APPROVED** [review]
LGTM

File: pkg/docker-engine/Dockerfile (hunk 1 above)
Comment (crazy-max) [reply pkg/docker-engine/Dockerfile:191]: This stage is to build static binaries like https://github.com/docker/docker-ce-packaging/pull/835

Comment (crazy-max) [reply pkg/docker-engine/Dockerfile:191]: In the future if I can rely on upstream Dockerfiles I could just use the right remote target and avoid redefining this stage.

**Review by neersighted — APPROVED** [review]

File: pkg/docker-engine/Dockerfile (hunk 1 above)
Comment (thaJeztah) [reply pkg/docker-engine/Dockerfile:191]: Oh! Yes for the static binaries it makes sense

**Pull request merged**